common-service/src/main/java/org/sunbird/global/ServiceBaseGlobal.java

```diff
@@ -0,0 +1,271 @@
+import play.mvc.Results;
+
+/**
+ * This class will work as a filter. All play request pass through this filter onCall method. Class
+ * also contains onStart method that will be call only once when application will be started.
+ *
+ * @author arvind.
+ */
+public class ServiceBaseGlobal extends BaseGlobal {
+
+  public static ProjectUtil.Environment env;
+
+  public static Map<String, Map<String, Object>> requestInfo = new HashMap<>();
+
+  public static String ssoPublicKey = "";
+  private static final String version = "v1";
+  private static final List<String> USER_UNAUTH_STATES =
+      Arrays.asList(JsonKey.UNAUTHORIZED, JsonKey.ANONYMOUS);
+
+  private class ActionWrapper extends Action.Simple {
+    public ActionWrapper(Action<?> action) {
+      this.delegate = action;
+    }
+
+    @Override
+    public Promise<Result> call(Http.Context ctx) throws java.lang.Throwable {
+      Promise<Result> result = null;
+      ctx.response().setHeader("Access-Control-Allow-Origin", "*");
+      // Verify request data like authentication
+      String message = RequestInterceptor.verifyRequestData(ctx);
+      // Set required parameters for telemetry event
+      intializeRequestInfo(ctx, message);
+      if (!USER_UNAUTH_STATES.contains(message)) {
+        ctx.flash().put(JsonKey.USER_ID, message);
+        ctx.flash().put(JsonKey.IS_AUTH_REQ, "false");
+        for (String uri : RequestInterceptor.getRestrictedUrlList()) {
+          if (ctx.request().path().contains(uri)) {
+            ctx.flash().put(JsonKey.IS_AUTH_REQ, "true");
+            break;
+          }
+        }
+        result = delegate.call(ctx);
+      } else if (JsonKey.UNAUTHORIZED.equals(message)) {
+        result =
+            onDataValidationError(
+                ctx.request(), message, ResponseCode.UNAUTHORIZED.getResponseCode());
+      } else {
+        result = delegate.call(ctx);
+      }
+      return result;
+    }
+  }
+
+  /**
+   * This method will be called on application start up. it will be called only time in it's life
+   * cycle.
+   *
+   * @param app Application
+   */
+  public void onStart(Application app) {
+    setEnvironment();
+    ProjectLogger.log("Server started.. with environment: " + env.name(), LoggerEnum.INFO.name());
+    ssoPublicKey = System.getenv(JsonKey.SSO_PUBLIC_KEY);
+    CassandraStartUpUtil.checkCassandraDbConnections(JsonKey.SUNBIRD);
+    CassandraStartUpUtil.checkCassandraDbConnections(JsonKey.SUNBIRD_PLUGIN);
+  }
+
+  /**
+   * This method will be called on each request.
+   *
+   * @param request represents the Play request object.
+   * @param actionMethod Method type , example - GET , POST etc.
+   * @return Action
+   */
+  @SuppressWarnings("rawtypes")
+  public Action onRequest(Request request, Method actionMethod) {
+
+    String messageId = request.getHeader(JsonKey.MESSAGE_ID);
+    if (StringUtils.isBlank(messageId)) {
+      UUID uuid = UUID.randomUUID();
+      messageId = uuid.toString();
```

Comment:
Combine into a single line of code.

messageId = UUID.randomUUID().toString();